cmake/tests.cmake

```diff
@@ -1,22 +1,20 @@
 function(make_tests_from_folder folder)
-
-  set (SINGLE_ARGS NAME)
-  set (MULTI_ARGS DEPENDS INCLUDES ARGUMENTS)
+  set(SINGLE_ARGS NAME)
+  set(MULTI_ARGS DEPENDS INCLUDES ARGUMENTS)
 
   cmake_parse_arguments(TEST "" "${SINGLE_ARGS}" "${MULTI_ARGS}" ${ARGN})
 
-  set (target "${TEST_NAME}")
+  set(target "${TEST_NAME}")
 
   message(STATUS "Create test ${target}")
 
-  file (GLOB_RECURSE SOURCES 
-        "${folder}/*.cpp"
-        "${folder}/*.hpp")
+  file(GLOB_RECURSE SOURCES 
+      "${folder}/*.cpp"
+      "${folder}/*.hpp"
+  )
 
   add_executable(${target} ${SOURCES})
-  target_link_libraries(${target} gtest ${TEST_DEPENDS})
+  target_link_libraries(${target} GTest::gtest_main ${TEST_DEPENDS})
```

Comment:
It worked fine without linking it to main in my experience. @NikitaZotov Can you explain this change please?

Reply:
Hmm, cmake can't find required symbols without specifying this target

`
Undefined symbols for architecture arm64:
[build]   "_main", referenced from:
[build]       <initial-undefines>
[build] ld: symbol(s) not found for architecture arm64
`

Reply:
I remember that you specified both gtest and gtest_main. Are you sure that you didn't have any problems here?